cmake/onnxruntime_webassembly.cmake

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+file(GLOB_RECURSE onnxruntime_wasm_src CONFIGURE_DEPENDS
+  "${ONNXRUNTIME_ROOT}/wasm/api.cc"
+)
+
+source_group(TREE ${REPO_ROOT} FILES ${onnxruntime_wasm_src})
+
+add_executable(onnxruntime_wasm
+  ${onnxruntime_wasm_src}
+)
+
+add_compile_definitions(
+  MLAS_NO_ONNXRUNTIME_THREADPOOL
+)
+
+string(APPEND CMAKE_CXX_FLAGS_DEBUG " -g4 --source-map-base http://localhost:8000/")
+
+target_compile_options(onnx PRIVATE -Wno-unused-parameter -Wno-unused-variable)
+
+# Override re2 compiler options to remove -pthread
+set_property(TARGET re2 PROPERTY COMPILE_OPTIONS )
+
+target_link_libraries(onnxruntime_wasm PRIVATE
+  protobuf::libprotobuf
```

Comment:
When possible, you may try protobuf-lite.  I guess you want something lighter and simpler.